### PR TITLE
nordic_nrf: Add ability to read FICR device ID from non-secure

### PR DESCRIPTION
**Pull request by joerchan**
Refactor soc_secure handling to not use TF-M directly in the header.
Move from nRF53 to common since nRF91 also supports TF-M.
This also allows for other adaption layers to be implemented for alternatives to TF-M.

Add soc_secure_read_deviceid function for reading the device ID from
FICR when executing in non-secure processing environment.

Remove the limitation in HWINFO_NRF not working in non-secure
configuration. Use the exposed soc_secure_read_deviceid function
that accesses the device ID through the secure services.

Allows the setting of ieee802154 EUI64 address in non-secure processing
environment by reading the FICR device ID through the secure service.

### CODE REVIEW

**Review by alexanderwachter — DISMISSED** [review]
+1 for device id

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -69,16 +70,15 @@ static struct nrf5_802154_data nrf5_data;
...
 #if defined(CONFIG_SOC_NRF5340_CPUAPP)
+#if defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+#error "NRF_UICR->OTP is not supported to read from non-secure"
+#else
 #define EUI64_ADDR (NRF_UICR->OTP)
+#endif /* CONFIG_TRUSTED_EXECUTION_NONSECURE */
 #else
 #define EUI64_ADDR (NRF_UICR->CUSTOMER)
 #endif /* CONFIG_SOC_NRF5340_CPUAPP */
 #else
```
Comment (jciupis) [inline drivers/ieee802154/ieee802154_nrf5.c:81]: This `#else` is unneeded.

**Review by jciupis — DISMISSED** [review]

**Review by jciupis — APPROVED** [review]

File: soc/arm/nordic_nrf/common/soc_secure.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc_secure.h>
+#include <errno.h>
+
+#include "nrf.h"
```
Comment (anangl) [inline soc/arm/nordic_nrf/common/soc_secure.c:10]: This is redundant.

File: soc/arm/nordic_nrf/common/soc_secure.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdint.h>
+#include <nrf.h>
```
Comment (anangl) [inline soc/arm/nordic_nrf/common/soc_secure.h:7]: This one as well.

**Review by anangl — APPROVED** [review]

**Pull request merged**